src/js/utils/MesosStateUtil.js

```diff
@@ -26,6 +26,19 @@ const MesosStateUtil = {
   },
 
   /**
+   * @param {{frameworks:array,completed_frameworks:array}} state
+   * @param {string} frameworkID
+   * @returns {{executors:Array, completed_executors:Array}|null} framework
+   */
+  getFramework(state, frameworkID) {
+    const {frameworks, completed_frameworks} = state;
+    return [].concat(frameworks, completed_frameworks).find(
+        function (framework) {
+          return framework.id === frameworkID;
```

Comment:
Make sure `framework` is defined, because `[].concat()` will introduce a `null` or `undefined` value if either of frameworks or completed_frameworks is missing.